--- a/chrome/browser/extensions/component_loader.cc	2018-05-01 
03:04:58.000000000 -0400
+++ b/chrome/browser/extensions/component_loader.cc.new	2018-05-01 
13:04:43.456738461 -0400
@@ -421,10 +421,6 @@
     return;
 #endif
 
-  AddWithNameAndDescription(
-      IDR_WEBSTORE_MANIFEST, base::FilePath(FILE_PATH_LITERAL("web_store")),
-      l10n_util::GetStringUTF8(IDS_WEBSTORE_NAME_STORE),
-      l10n_util::GetStringUTF8(IDS_WEBSTORE_APP_DESCRIPTION));
 }
 
 scoped_refptr<const Extension> ComponentLoader::CreateExtension(
@@ -467,11 +463,6 @@
   }
 #else  // defined(OS_CHROMEOS)
   DCHECK(!skip_session_components);
-#if BUILDFLAG(ENABLE_PRINTING)
-  // Cloud Print component app. Not required on Chrome OS.
-  Add(IDR_CLOUDPRINT_MANIFEST,
-      base::FilePath(FILE_PATH_LITERAL("cloud_print")));
-#endif  // BUILDFLAG(ENABLE_PRINTING)
 #endif  // defined(OS_CHROMEOS)
 
   if (!skip_session_components) {
@@ -548,12 +539,6 @@
     AddHangoutServicesExtension();
     AddImageLoaderExtension();
 
-    bool install_feedback = enable_background_extensions_during_testing;
-#if defined(GOOGLE_CHROME_BUILD)
-    install_feedback = true;
-#endif  // defined(GOOGLE_CHROME_BUILD)
-    if (install_feedback)
-      Add(IDR_FEEDBACK_MANIFEST, base::FilePath(FILE_PATH_LITERAL("feedback")));
   }
 
 #if defined(OS_CHROMEOS)
