--- a/chrome/browser/extensions/webstore_installer.cc	2017-10-04 18:07:34.000000000 -0400
+++ b/chrome/browser/extensions/webstore_installer.cc	2017-10-08 18:49:29.349041815 -0400
@@ -580,6 +580,7 @@
 void WebstoreInstaller::DownloadCrx(
     const std::string& extension_id,
     InstallSource source) {
+#if 0
   download_url_ = GetWebstoreInstallURL(extension_id, source);
   MaybeAppendAuthUserParameter(approval_->authuser, &download_url_);
 
@@ -602,6 +603,7 @@
       GetExtensionFileTaskRunner().get(), FROM_HERE,
       base::BindOnce(&GetDownloadFilePath, download_directory, extension_id),
       base::BindOnce(&WebstoreInstaller::StartDownload, this, extension_id));
+#endif /* 0 */
 }
 
 // http://crbug.com/165634
@@ -749,6 +751,7 @@
 }
 
 void WebstoreInstaller::StartCrxInstaller(const DownloadItem& download) {
+#if 0
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
   DCHECK(!crx_installer_.get());
 
@@ -771,6 +774,7 @@
   crx_installer_->set_allow_silent_install(true);
 
   crx_installer_->InstallCrx(download.GetFullPath());
+#endif /* 0 */
 }
 
 void WebstoreInstaller::ReportFailure(const std::string& error,
