--- a/chrome/browser/ui/browser_ui_prefs.cc	2018-06-21 
21:22:12.661091321 -0400
+++ b/chrome/browser/ui/browser_ui_prefs.cc.new	2018-06-21 
21:24:27.148632573 -0400
@@ -74,7 +74,7 @@
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterStringPref(prefs::kCloudPrintEmail, std::string());
   registry->RegisterBooleanPref(prefs::kCloudPrintProxyEnabled, true);
-  registry->RegisterBooleanPref(prefs::kCloudPrintSubmitEnabled, true);
+  registry->RegisterBooleanPref(prefs::kCloudPrintSubmitEnabled, false);
   registry->RegisterDictionaryPref(prefs::kBrowserWindowPlacement);
   registry->RegisterDictionaryPref(prefs::kBrowserWindowPlacementPopup);
   registry->RegisterDictionaryPref(prefs::kAppWindowPlacement);
@@ -87,10 +87,10 @@
                                 false);
 #endif
   // TODO(guoweis): Remove next 2 options at M50.
-  registry->RegisterBooleanPref(prefs::kWebRTCMultipleRoutesEnabled, true);
-  registry->RegisterBooleanPref(prefs::kWebRTCNonProxiedUdpEnabled, true);
+  registry->RegisterBooleanPref(prefs::kWebRTCMultipleRoutesEnabled, false);
+  registry->RegisterBooleanPref(prefs::kWebRTCNonProxiedUdpEnabled, false);
   registry->RegisterStringPref(prefs::kWebRTCIPHandlingPolicy,
-                               content::kWebRTCIPHandlingDefault);
+                               content::kWebRTCIPHandlingDisableNonProxiedUdp);
   registry->RegisterStringPref(prefs::kWebRTCUDPPortRange, std::string());
 
   // Dictionaries to keep track of default tasks in the file browser.
