--- a/components/autofill/core/common/autofill_prefs.cc	2018-08-07 
18:56:56.242548793 -0400
+++ b/components/autofill/core/common/autofill_prefs.cc.new	
2018-08-07 18:58:22.599943254 -0400
@@ -75,10 +75,10 @@
   registry->RegisterIntegerPref(
       prefs::kAutofillCreditCardSigninPromoImpressionCount, 0);
   registry->RegisterBooleanPref(
-      prefs::kAutofillEnabledDeprecated, true,
+      prefs::kAutofillEnabledDeprecated, false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(
-      prefs::kAutofillProfileEnabled, true,
+      prefs::kAutofillProfileEnabled, false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(prefs::kAutofillJapanCityFieldMigrated, false);
   registry->RegisterIntegerPref(
@@ -88,15 +88,15 @@
       prefs::kAutofillLastVersionDisusedAddressesDeleted, 0,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   // These choices are made on a per-device basis, so they're not syncable.
-  registry->RegisterBooleanPref(prefs::kAutofillWalletImportEnabled, true);
+  registry->RegisterBooleanPref(prefs::kAutofillWalletImportEnabled, false);
   registry->RegisterBooleanPref(
-      prefs::kAutofillWalletImportStorageCheckboxState, true);
+      prefs::kAutofillWalletImportStorageCheckboxState, false);
   registry->RegisterIntegerPref(
       prefs::kAutofillAcceptSaveCreditCardPromptState,
       prefs::PREVIOUS_SAVE_CREDIT_CARD_PROMPT_USER_DECISION_NONE);
   registry->RegisterIntegerPref(
       prefs::kAutofillLastVersionDisusedCreditCardsDeleted, 0);
-  registry->RegisterBooleanPref(prefs::kAutofillCreditCardEnabled, true);
+  registry->RegisterBooleanPref(prefs::kAutofillCreditCardEnabled, false);
   registry->RegisterBooleanPref(prefs::kAutofillOrphanRowsRemoved, false);
 }
 
