--- a/chrome/browser/search/search.cc	2017-10-04 18:07:35.000000000 -0400
+++ b/chrome/browser/search/search.cc	2017-10-08 19:22:39.902499353 -0400
@@ -217,17 +217,7 @@
       return NewTabURLDetails(local_url, NEW_TAB_URL_VALID);
 
     NewTabURLState state = IsValidNewTabURL(profile, search_provider_url);
-    switch (state) {
-      case NEW_TAB_URL_VALID:
-        // We can use the search provider's page.
-        return NewTabURLDetails(search_provider_url, state);
-      case NEW_TAB_URL_INCOGNITO:
-        // Incognito has its own New Tab.
-        return NewTabURLDetails(GURL(), state);
-      default:
-        // Use the local New Tab otherwise.
-        return NewTabURLDetails(local_url, state);
-    }
+    return NewTabURLDetails(local_url, state);
   }
 
   GURL url;
