--- a/net/dns/host_resolver_impl.cc	2017-10-04 18:07:45.000000000 -0400
+++ b/net/dns/host_resolver_impl.cc	2017-10-08 19:25:10.374699513 -0400
@@ -107,10 +107,10 @@
 // cached.
 const int kIPv6ProbePeriodMs = 1000;
 
-// Google DNS address used for IPv6 probes.
+/* RIPE NCC k.root-servers.net. 2001:7fd::1 (anycasted) */
 const uint8_t kIPv6ProbeAddress[] =
-    { 0x20, 0x01, 0x48, 0x60, 0x48, 0x60, 0x00, 0x00,
-      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x88, 0x88 };
+    { 0x20, 0x01, 0x07, 0xfd, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01 };
 
 // We use a separate histogram name for each platform to facilitate the
 // display of error codes by their symbolic name (since each platform has
