--- a/tools/gn/bootstrap/bootstrap.py	2018-05-20 13:36:48.004889745 
-0400
+++ b/tools/gn/bootstrap/bootstrap.py.new	2018-05-20 
14:08:48.286761101 -0400
@@ -602,6 +602,7 @@
       'base/task_scheduler/sequence_sort_key.cc',
       'base/task_scheduler/task.cc',
       'base/task_scheduler/task_scheduler.cc',
+      'base/task_scheduler/service_thread.cc',
       'base/task_scheduler/task_scheduler_impl.cc',
       'base/task_scheduler/task_tracker.cc',
       'base/task_scheduler/task_traits.cc',
