diff --git a/src/api.cc b/src/api.cc
index e7cc4f8..15bbed4 100644
--- a/v8/src/api.cc
+++ b/v8/src/api.cc
@@ -10471,9 +10471,9 @@
 #pragma clang diagnostic ignored "-Wdeprecated"
 #endif
   return !this->AdvanceTracing(
-      deadline_in_ms,
-      AdvanceTracingActions(isinf(deadline_in_ms) ? FORCE_COMPLETION
-                                                  : DO_NOT_FORCE_COMPLETION));
+      deadline_in_ms, AdvanceTracingActions(std::isinf(deadline_in_ms)
+                                                ? FORCE_COMPLETION
+                                                : DO_NOT_FORCE_COMPLETION));
 #if __clang__
 #pragma clang diagnostic pop
 #endif
