--- a/tools/gn/bootstrap/bootstrap.py	2018-05-23 18:08:29.819531735 
-0400
+++ b/tools/gn/bootstrap/bootstrap.py.new	2018-05-23 
18:26:03.575318658 -0400
@@ -649,7 +649,6 @@
       'base/trace_event/memory_peak_detector.cc',
       'base/trace_event/memory_usage_estimator.cc',
       'base/trace_event/process_memory_dump.cc',
-      'base/trace_event/sharded_allocation_register.cc',
       'base/trace_event/trace_buffer.cc',
       'base/trace_event/trace_config.cc',
       'base/trace_event/trace_config_category_filter.cc',
