--- a/pdf/pdfium/pdfium_mem_buffer_file_write.cc	2018-08-27 
06:09:45.000000000 -0400
+++ b/pdf/pdfium/pdfium_mem_buffer_file_write.cc.new	2018-08-27 
20:55:22.415727873 -0400
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <cstdint>
+
 #include "pdf/pdfium/pdfium_mem_buffer_file_write.h"
 
 namespace chrome_pdf {
