diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index 6e82635..75d7d5c 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1579,6 +1579,7 @@
     "//content/public/browser",
     "//ipc",
     "//sql",
+    "//chrome/common:service_process_mojom",
   ]
   deps = [
     ":active_use_util",
@@ -3136,8 +3137,6 @@
       "notifications/screen_lock_notification_blocker.cc",
       "notifications/screen_lock_notification_blocker.h",
       "platform_util.cc",
-      "policy/cloud/machine_level_user_cloud_policy_helper.cc",
-      "policy/cloud/machine_level_user_cloud_policy_helper.h",
       "policy/cloud/user_policy_signin_service.cc",
       "policy/cloud/user_policy_signin_service.h",
       "profiles/avatar_menu.cc",
