--- a/chrome/browser/signin/signin_promo.cc	2017-10-11 15:10:23.000000000 -0400
+++ b/chrome/browser/signin/signin_promo.cc	2017-10-15 10:44:43.331167859 -0400
@@ -383,7 +383,7 @@
 void RegisterProfilePrefs(
     user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterIntegerPref(prefs::kSignInPromoStartupCount, 0);
-  registry->RegisterBooleanPref(prefs::kSignInPromoUserSkipped, false);
+  registry->RegisterBooleanPref(prefs::kSignInPromoUserSkipped, true);
   registry->RegisterBooleanPref(prefs::kSignInPromoShowOnFirstRunAllowed, true);
   registry->RegisterBooleanPref(prefs::kSignInPromoShowNTPBubble, false);
   registry->RegisterIntegerPref(prefs::kDiceSigninUserMenuPromoCount, 0);
