--- a/components/safe_browsing/features.cc	2017-10-11 15:10:28.000000000 -0400
+++ b/components/safe_browsing/features.cc	2017-10-15 11:00:50.754740977 -0400
@@ -47,7 +47,7 @@
 
 const base::Feature kGoogleBrandedPhishingWarning{
     "PasswordProtectionGoogleBrandedPhishingWarning",
-    base::FEATURE_ENABLED_BY_DEFAULT};
+    base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kThreatDomDetailsTagAndAttributeFeature{
     "ThreatDomDetailsTagAttributes", base::FEATURE_DISABLED_BY_DEFAULT};
@@ -57,7 +57,7 @@
     base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kDispatchSafetyNetCheckOffThread{
-    "DispatchSafetyNetCheckOffThread", base::FEATURE_ENABLED_BY_DEFAULT};
+    "DispatchSafetyNetCheckOffThread", base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kAppendRecentNavigationEvents{
     "AppendRecentNavigationEvents", base::FEATURE_DISABLED_BY_DEFAULT};
@@ -79,13 +79,13 @@
 } kExperimentalFeatures[]{
     {&kAdSamplerCollectButDontSendFeature, false},
     {&kAdSamplerTriggerFeature, false},
-    {&kAppendRecentNavigationEvents, true},
-    {&kCheckByURLLoaderThrottle, true},
+    {&kAppendRecentNavigationEvents, false},
+    {&kCheckByURLLoaderThrottle, false},
     {&kDispatchSafetyNetCheckOffThread, false},
-    {&kEnterprisePasswordProtectionV1, true},
-    {&kGaiaPasswordReuseReporting, true},
+    {&kEnterprisePasswordProtectionV1, false},
+    {&kGaiaPasswordReuseReporting, false},
     {&kGoogleBrandedPhishingWarning, false},
-    {&kInspectDownloadedRarFiles, true},
+    {&kInspectDownloadedRarFiles, false},
     {&kThreatDomDetailsTagAndAttributeFeature, false},
     {&kTriggerThrottlerDailyQuotaFeature, false},
 };
