--- a/components/safe_browsing/browser/threat_details_cache.cc	2017-10-11 15:10:28.000000000 -0400
+++ b/components/safe_browsing/browser/threat_details_cache.cc	2017-10-15 11:04:58.374841599 -0400
@@ -113,7 +113,8 @@
   // Only from cache, and don't save cookies.
   resource_request->load_flags = net::LOAD_ONLY_FROM_CACHE |
                                  net::LOAD_SKIP_CACHE_VALIDATION |
-                                 net::LOAD_DO_NOT_SAVE_COOKIES;
+                                 net::LOAD_DO_NOT_SAVE_COOKIES |
+                                 net::LOAD_DO_NOT_SEND_COOKIES;
   current_load_ = network::SimpleURLLoader::Create(std::move(resource_request),
                                                    traffic_annotation);
   current_load_->DownloadToStringOfUnboundedSizeUntilCrashAndDie(
