--- a/chrome/browser/ui/startup/startup_tab_provider.cc	2017-10-18 17:59:46.753324198 -0400
+++ b/chrome/browser/ui/startup/startup_tab_provider.cc	2017-10-18 17:59:14.879923757 -0400
@@ -194,13 +194,6 @@
 StartupTabs StartupTabProviderImpl::GetStandardOnboardingTabsForState(
     const StandardOnboardingTabsParams& params) {
   StartupTabs tabs;
-  if (CanShowWelcome(params.is_signin_allowed, params.is_supervised_user,
-                     params.is_force_signin_enabled) &&
-      ShouldShowWelcomeForOnboarding(params.has_seen_welcome_page,
-                                     params.is_signed_in,
-                                     params.is_signin_in_progress)) {
-    tabs.emplace_back(GetWelcomePageUrl(!params.is_first_run), false);
-  }
   return tabs;
 }
 
