--- a/chrome/browser/media/router/media_router_feature.cc	
2018-04-10 12:49:19.136800802 -0400
+++ b/chrome/browser/media/router/media_router_feature.cc	
2018-04-10 12:55:38.039057728 -0400
@@ -11,26 +11,26 @@
 #include "extensions/buildflags/buildflags.h"
 #include "ui/base/ui_features.h"
 
-#if defined(OS_ANDROID) || BUILDFLAG(ENABLE_EXTENSIONS)
+
 #include "chrome/common/pref_names.h"
 #include "components/prefs/pref_service.h"
 #include "components/user_prefs/user_prefs.h"
-#endif  // defined(OS_ANDROID) || BUILDFLAG(ENABLE_EXTENSIONS)
 
-#if !defined(OS_ANDROID)
+
+
 #include "components/prefs/pref_registry_simple.h"
-#endif
+
 
 namespace media_router {
 
-#if !defined(OS_ANDROID)
+
 // Controls if browser side DIAL sink query is enabled.
 const base::Feature kEnableDialSinkQuery{"EnableDialSinkQuery",
                                          base::FEATURE_DISABLED_BY_DEFAULT};
 
 // Controls if browser side Cast device discovery is enabled.
 const base::Feature kEnableCastDiscovery{"EnableCastDiscovery",
-                                         base::FEATURE_ENABLED_BY_DEFAULT};
+                                         base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kCastMediaRouteProvider{"CastMediaRouteProvider",
                                             base::FEATURE_DISABLED_BY_DEFAULT};
@@ -38,9 +38,9 @@
 // Controls if local media casting is enabled.
 const base::Feature kEnableCastLocalMedia{"EnableCastLocalMedia",
                                           base::FEATURE_DISABLED_BY_DEFAULT};
-#endif
 
-#if defined(OS_ANDROID) || BUILDFLAG(ENABLE_EXTENSIONS)
+
+
 namespace {
 const PrefService::Preference* GetMediaRouterPref(
     content::BrowserContext* context) {
@@ -48,10 +48,10 @@
       ::prefs::kEnableMediaRouter);
 }
 }  // namespace
-#endif  // defined(OS_ANDROID) || BUILDFLAG(ENABLE_EXTENSIONS)
+
 
 bool MediaRouterEnabled(content::BrowserContext* context) {
-#if defined(OS_ANDROID) || BUILDFLAG(ENABLE_EXTENSIONS)
+
   const PrefService::Preference* pref = GetMediaRouterPref(context);
   // Only use the pref value if it set from a mandatory policy.
   if (pref->IsManaged() && !pref->IsDefaultValue()) {
@@ -62,13 +62,10 @@
 
   // The component extension cannot be loaded in guest sessions.
   // TODO(crbug.com/756243): Figure out why.
-  return !Profile::FromBrowserContext(context)->IsGuestSession();
-#else  // !(defined(OS_ANDROID) || BUILDFLAG(ENABLE_EXTENSIONS))
   return false;
-#endif  // defined(OS_ANDROID) || BUILDFLAG(ENABLE_EXTENSIONS)
 }
 
-#if !defined(OS_ANDROID)
+
 void RegisterLocalStatePrefs(PrefRegistrySimple* registry) {
   registry->RegisterBooleanPref(prefs::kMediaRouterCastAllowAllIPs, false,
                                 PrefRegistry::PUBLIC);
@@ -113,12 +110,7 @@
 // Returns true if the presentation receiver window for local media casting is
 // available on the current platform.
 bool PresentationReceiverWindowEnabled() {
-#if defined(OS_MACOSX) && !BUILDFLAG(MAC_VIEWS_BROWSER)
-  return false;
-#else
   return true;
-#endif
 }
-#endif  // !defined(OS_ANDROID)
 
 }  // namespace media_router
