--- a/chrome/browser/media/webrtc/webrtc_log_uploader.cc	2018-08-18 19:44:26.191017194 -0400
+++ b/chrome/browser/media/webrtc/webrtc_log_uploader.cc.new	2018-08-18 19:44:44.101529263 -0400
@@ -117,30 +117,9 @@
   DCHECK(meta_data.get());
   DCHECK(!upload_done_data.log_path.empty());
 
-  std::string compressed_log;
-  CompressLog(&compressed_log, log_buffer.get());
 
-  std::string local_log_id;
 
-  if (base::PathExists(upload_done_data.log_path)) {
-    webrtc_logging::DeleteOldWebRtcLogFiles(upload_done_data.log_path);
-
-    local_log_id = base::NumberToString(base::Time::Now().ToDoubleT());
-    base::FilePath log_file_path =
-        upload_done_data.log_path.AppendASCII(local_log_id)
-            .AddExtension(FILE_PATH_LITERAL(".gz"));
-    WriteCompressedLogToFile(compressed_log, log_file_path);
-
-    base::FilePath log_list_path =
-        webrtc_logging::TextLogList::GetWebRtcLogListFileForDirectory(
-            upload_done_data.log_path);
-    AddLocallyStoredLogInfoToUploadListFile(log_list_path, local_log_id);
-  }
-
-  WebRtcLogUploadDoneData upload_done_data_with_log_id = upload_done_data;
-  upload_done_data_with_log_id.local_log_id = local_log_id;
-  PrepareMultipartPostData(compressed_log, std::move(meta_data),
-                           upload_done_data_with_log_id);
+  NotifyUploadDone(net::HTTP_OK, "", upload_done_data);
 }
 
 void WebRtcLogUploader::PrepareMultipartPostData(
@@ -151,6 +130,9 @@
   DCHECK(!compressed_log.empty());
   DCHECK(meta_data.get());
 
+  NotifyUploadDone(net::HTTP_OK, "", upload_done_data);
+  return;
+
   std::unique_ptr<std::string> post_data(new std::string());
   SetupMultipart(post_data.get(), compressed_log,
                  upload_done_data.incoming_rtp_dump,
