--- a/google_apis/gaia/gaia_auth_fetcher.cc	2017-10-17 15:10:41.000000000 -0400
+++ b/google_apis/gaia/gaia_auth_fetcher.cc	2017-10-18 10:50:58.344372939 -0400
@@ -222,35 +222,6 @@
     int load_flags,
     const net::NetworkTrafficAnnotationTag& traffic_annotation) {
   DCHECK(!fetch_pending_) << "Tried to fetch two things at once!";
-  fetcher_ = net::URLFetcher::Create(
-      0, gaia_gurl, body.empty() ? net::URLFetcher::GET : net::URLFetcher::POST,
-      this, traffic_annotation);
-  fetcher_->SetRequestContext(getter_);
-  fetcher_->SetUploadData("application/x-www-form-urlencoded", body);
-  gaia::MarkURLFetcherAsGaia(fetcher_.get());
-
-  VLOG(2) << "Gaia fetcher URL: " << gaia_gurl.spec();
-  VLOG(2) << "Gaia fetcher headers: " << headers;
-  VLOG(2) << "Gaia fetcher body: " << body;
-
-  // The Gaia token exchange requests do not require any cookie-based
-  // identification as part of requests.  We suppress sending any cookies to
-  // maintain a separation between the user's browsing and Chrome's internal
-  // services.  Where such mixing is desired (MergeSession or OAuthLogin), it
-  // will be done explicitly.
-  fetcher_->SetLoadFlags(load_flags);
-
-  // Fetchers are sometimes cancelled because a network change was detected,
-  // especially at startup and after sign-in on ChromeOS. Retrying once should
-  // be enough in those cases; let the fetcher retry up to 3 times just in case.
-  // http://crbug.com/163710
-  fetcher_->SetAutomaticallyRetryOnNetworkChanges(3);
-
-  if (!headers.empty())
-    fetcher_->SetExtraRequestHeaders(headers);
-
-  fetch_pending_ = true;
-  fetcher_->Start();
 }
 
 // static
