--- a/content/browser/service_worker/service_worker_write_to_cache_job.cc	2017-10-11 15:10:30.000000000 -0400
+++ b/content/browser/service_worker/service_worker_write_to_cache_job.cc	2017-10-17 18:09:59.259395092 -0400
@@ -226,8 +226,7 @@
               "stopped."
             chrome_policy {
               DefaultCookiesSetting {
-                policy_options {mode: MANDATORY}
-                DefaultCookiesSetting: 2
+                DefaultCookiesSetting: 4
               }
             }
           })");
