--- a/third_party/skia/src/core/SkColorData.h	2018-11-03 
23:22:37.639351789 -0400
+++ b/third_party/skia/src/core/SkColorData.h.new	2018-11-03 
23:23:58.753083522 -0400
@@ -10,8 +10,8 @@
 
 #include "SkColor.h"
 #include "SkColorPriv.h"
-#include "SkNx.h"
-#include "SkTo.h"
+#include "../../include/private/SkNx.h"
+#include "../../include/private/SkTo.h"
 
 ////////////////////////////////////////////////////////////////////////////////////////////
 // Convert a 16bit pixel to a 32bit pixel
