--- a/third_party/blink/public/common/origin_policy/origin_policy.h	2018-06-27 06:33:15.000000000 -0400
+++ b/third_party/blink/public/common/origin_policy/origin_policy.h.new	2018-06-30 20:02:17.704399964 -0400
@@ -6,6 +6,8 @@
 #define THIRD_PARTY_BLINK_PUBLIC_COMMON_ORIGIN_POLICY_ORIGIN_POLICY_H_
 
 #include <string>
+#include <memory>
+
 
 #include "base/strings/string_piece.h"
 #include "third_party/blink/common/common_export.h"

--- a/third_party/blink/common/origin_policy/origin_policy_parser.h	2018-06-27 06:33:14.000000000 -0400
+++ b/third_party/blink/common/origin_policy/origin_policy_parser.h.new	2018-06-30 20:03:16.238115271 -0400
@@ -9,6 +9,8 @@
 
 #include <string>
 #include "base/macros.h"
+#include <memory>
+
 
 namespace blink {
 

--- a/third_party/blink/common/origin_policy/origin_policy.cc	2018-07-02 23:10:04.000000000 -0400
+++ b/third_party/blink/common/origin_policy/origin_policy.cc.new	2018-07-03 20:29:33.191695083 -0400
@@ -3,6 +3,7 @@
 // found in the LICENSE file.
 
 #include "third_party/blink/public/common/origin_policy/origin_policy.h"
+#include <memory>
 
 #include "third_party/blink/common/origin_policy/origin_policy_parser.h"
 
--- a/third_party/blink/common/origin_policy/origin_policy_parser.h	2018-07-02 23:10:04.000000000 -0400
+++ b/third_party/blink/common/origin_policy/origin_policy_parser.h.new	2018-07-03 20:29:34.198369576 -0400
@@ -8,6 +8,7 @@
 #include "third_party/blink/public/common/origin_policy/origin_policy.h"
 
 #include <string>
+#include <memory>
 #include "base/macros.h"
 
 namespace blink {

--- a/third_party/webrtc/modules/audio_processing/aec3/reverb_model_estimator.cc	2018-07-05 03:24:02.000000000 -0400
+++ b/third_party/webrtc/modules/audio_processing/aec3/reverb_model_estimator.cc.new	2018-07-05 17:57:39.152549112 -0400
@@ -15,6 +15,7 @@
 #include <memory>
 #include <numeric>
 
+#include <math.h>
 #include "api/array_view.h"
 #include "api/audio/echo_canceller3_config.h"
 #include "modules/audio_processing/aec3/aec3_common.h"
